ci/do_ci.sh

```diff
@@ -91,8 +91,8 @@ function cp_binary_for_image_build() {
   mkdir -p "${BASE_TARGET_DIR}"/"${TARGET_DIR}"_stripped
   strip "${FINAL_DELIVERY_DIR}"/envoy -o "${BASE_TARGET_DIR}"/"${TARGET_DIR}"_stripped/envoy
 
-  # Copy for azp which doesn't preserve permissions, creating a tar archive
-  tar czf "${ENVOY_BUILD_DIR}"/"${EXE_NAME}"_binary.tar.gz -C "${BASE_TARGET_DIR}" "${TARGET_DIR}" "${TARGET_DIR}"_stripped
+  # Copy for azp which doesn't preserve permissions, creating a tar archive only when release
+  [[ "${BINARY_TYPE}" == "release" ]] && tar czf "${ENVOY_BUILD_DIR}"/"${EXE_NAME}"_binary.tar.gz -C "${BASE_TARGET_DIR}" "${TARGET_DIR}" "${TARGET_DIR}"_stripped
```

Comment:
looking through the code more thoroughly - if we only want to create the tarball in azp then it seems like the next condition below - ie `[[ -z "${BUILD_REASON}" ]]` can be used

how about just moving the tar line into that condition ? (perhaps adding `if .. fi` to make it more readable)

Reply:
Sure.

Reply:
i dont see the change

/wait

Reply:
sorry for push issue. now you can see it.